src/Orleans/Providers/IStorageProvider.cs

```diff
@@ -70,7 +70,7 @@ public interface IStorageProvider : IProvider
     /// <summary>
     /// Interface to be optionally implemented by storage providers to return richer exception details.
     /// </summary>
-    public interface IExceptionDecoder
+    public interface IStorageExceptionDecoder
```

Comment:
I thought this could be used as a more general pattern beyond storage, e.g. for stream providers as well. So I gave it a more generic name. Maybe keep it as `IExceptionDecoder` but move out of IStorageProvider.cs?